### ISSUE TITLE
save changes to dandiset.yaml last

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandisets#143]
ATM, in case of crashes, like whenever some sha256 is not known for an asset, we seems to end up in a dirty dataset state

```
dandi@drogon:/mnt/backup/dandi/dandisets$ git status
On branch draft
Your branch is up to date with 'github/draft'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

        modified:   000026 (modified content)

no changes added to commit (use "git add" and/or "git commit -a")
dandi@drogon:/mnt/backup/dandi/dandisets$ cd 000026/
dandi@drogon:/mnt/backup/dandi/dandisets/000026$ git status
On branch draft
Your branch is up to date with 'github/draft'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   dandiset.yaml
```

since it seems changes to dandiset.yaml are saved first.  Would it be easy to save them  last, right before calling datalad's save on the dandiset?  Then there would be a better chance to remain in a clean state for subsequent cron job to possibly succeed.

**Comment by jwodder** [comment]
@yarikoptic Doing this might cause problems when a Dandiset with multiple versions is being backed up, as then the intermediate commits would lack a `dandiset.yaml`.  I can't run the tests to be sure, due to https://github.com/dandi/dandi-archive/issues/961.

**Comment by yarikoptic** [comment]
> @yarikoptic Doing this might cause problems when a Dandiset with multiple versions is being backed up, as then the intermediate commits would lack a `dandiset.yaml`.

indeed a concern -- would need somehow to ensure providing that updated `dandiset.yaml` at that point.  Ok, let's postpone until  we resolve that issue.

**Comment by yarikoptic** [comment]
I believe that original https://github.com/dandi/dandi-archive/issues/961 is addressed, please check - may be testing  is ready and this issue is unblocked??
